vignettes/cookbook.Rmd

```diff
@@ -36,68 +37,72 @@ Line graphs are useful for showing change over time, and are well-suited for com
   mutate(indexed = grp / first(grp) - 1)
 
 
-# create line plot
-p <- ggplot(data = df, mapping = aes(
-  x = year, y = indexed,
-  color = fct_reorder2(area, year, indexed),    # order legend by last `indexed` value in last `year`
-  label = scales::percent(indexed, accuracy = 1))) +   # format labels as percents with no decimals
+# Create line plot
+p <- ggplot(
+  data = df,
+  mapping = aes(
+    x = year,
+    y = indexed,
+    color = fct_reorder2(area, year, indexed),  # Order legend by last `indexed` value in last `year`
+    label = scales::percent(indexed, accuracy = 1)  # Format labels as percents with no decimals
+  )
+) +
   
-  # add geoms, making sure to keep recessions under lines
+  # Add geoms, making sure to keep recessions under lines
   geom_recessions(ymin = 0, ymax = .5) +
   geom_line(lineend = "round") +
   geom_text_lastonly() +
   
-  # establish an attractive Y scale
+  # Establish an attractive Y scale
   scale_y_continuous(
-    labels = scales::label_percent(accuracy = 1),      # Format y-axis numbers with commas
-    expand = expansion(add = 0)) +                     # No padding beyond specified limits
+    labels = scales::label_percent(accuracy = 1),
+    expand = expansion(add = 0)  # No padding beyond specified limits
+  ) +
   
-  # establish an attractive X scale
+  # Establish an attractive X scale
   scale_x_continuous(
-    breaks = seq(min(df$year), max(df$year), by = 2),  # Set breaks to every other year
-    expand = expansion(add = c(0.5, 1.25))) +          # Set custom padding for L and R side of axis 
+    breaks = seq(from = min(df$year), to = max(df$year), by = 2),  # Set breaks to every other year
+    expand = expansion(add = c(0.5, 1.25))  # Set custom padding for L/R sides of axis 
+  ) +
   
-  # Apply CMAP theme
+  # Apply CMAP theme and color palette
   theme_cmap(
-    overrides = list(margin_panel_r = 3)  # shrink distance btwn gridlines ending and plotting extent
-  )
+    overrides = list(margin_panel_r = 3)  # Shrink distance btwn gridlines ending and plotting extent
+  ) +
+  cmap_color_discrete(palette = "legislation")
   
 
 # Create final plot
 finalize_plot(
-  p, mode = "png", filename = "./cookbook_line.png",
+  p,
+  title = "Cumulative growth in real gross regional product, 2001-19",
+  caption = "Note: Index year 2001.<br><br>
+    Source: Chicago Metropolitan Agency for Planning analysis of U.S. Bureau
+    of Economic Analysis and Economic Modeling Specialists International data
+    (Emsi 2020.4).",
   
-  # modify default sizes a touch
+  # Tweak plot layout
   height = 4.5,
   title_width = 2.6,
-  
-  # use some <br> tags to get the title and caption to match perfectly
-  title = "Cumulative growth<br>in real gross regional product, 2001-19",
-  caption = "Note: Indext year 2001. <br><br> Source: Chicago Metropolitan
-    Agency for Planning analysis of U.S. Bureau <br>of Economic Analysis and 
-    Economic Modeling Specialists International <br>data (Emsi 2020.4).",
```

Comment:
@matthewstern FYI I removed the mid-sentence `<br>`s. While I appreciate that they were adding line breaks in the original locations, it doesn't quite work as intended: 
<img width="220" alt="Screen Shot 2020-11-22 at 2 01 39 PM" src="https://user-images.githubusercontent.com/4383272/99924703-e9c23180-2d00-11eb-9eea-3da7a77c1062.png">

I don't think we should be encouraging people to use `<br>` tags unless truly necessary, anyway (e.g. between note and caption). It just feels like we're trying too hard to exactly replicate the original.